### PR TITLE
Problem Solving Proposal

### PR DESCRIPTION
**Pull request by AlexDaniel**
Please leave feedback.

Resolves https://github.com/perl6/problem-solving/issues/1.

[Rendered document can be seen here.](https://github.com/perl6/problem-solving/blob/00000001/README.md)

TODOs:
- [x] ~ask TimToady if they can be added as a responsible dev to the `language` label~ – they can add themselves later
- [x] ~find people for labels with no responsible devs~ see https://github.com/perl6/problem-solving/pull/2#issuecomment-467207869
- [x] describe directory structure and naming conventions of the documents

### CODE REVIEW

**Comment by AlexDaniel** [comment]
In order to reduce the issue spam required to populate the list of reviewers, please simply react with :eyes: to this comment and I'll add you to the list of reviewers. If you have a commit bit, you should probably be on that list.

**Comment by AlexDaniel** [comment]
@JJ, @coke, documentation label currently has no responsible devs, please consider volunteering.

@ugexe same goes for `modules`, though I'm not sure what'd be the scope of that label. If you have some ideas, let me know. That said, there's no need to create all of the labels ahead of time, so we can delay some of that stuff until at least one ticket is created.

**Comment by JJ** [comment]
Yep, count me in for documentation.

File: README.md [hunk 1]
```diff
@@ -1 +1,159 @@
...
+* `language` – changes to the Perl 6 language
+  * `jnthn`
+* `rakudo` – big changes to rakudo
+  * `jnthn`
+* `moarvm` – big changes to moarvm
+  * `jnthn`
+* `modules` – ecosystem-related issues and META6.json
+  * 🏜 (please apply)
+* `documentation`
+  * 🏜 (please apply)
```
Comment (JJ) [inline README.md]: Please add me here.

**Comment by JJ** [comment]
I'd like to participate in modules also.

**Comment by ugexe** [comment]
I agree we need a categorization of our SME ([Subject Matter Experts](https://en.wikipedia.org/wiki/Subject-matter_expert#Software)). This should not be done in an opt-in way -- such a list should be available as a resource regardless. If some area is stuck because there is no associated SME or that SME is not participating then there should probably not be any significant changes made until someone learns the area to an acceptable degree such as to become an acting SME.

**Comment by AlexDaniel** [comment]
> This should not be done in an opt-in way

I'd love to agree but I don't see how we can force people to do work. What's the alternative?

**Comment by ugexe** [comment]
> If some area is stuck because there is no associated SME or that SME is not participating then there should probably not be any significant changes made until someone learns the area to an acceptable degree such as to become an acting SME.

So there is no forcing the SME to do anything. The only time force is required is when the person making a proposal has no SME participating -- in this case the proposer is forced to learn significantly more to become a SME if they wish to further their proposal.

**Comment by AlexDaniel** [comment]
@ugexe OK, I see. That makes sense, thank you.

**Comment by AlexDaniel** [comment]
This PR is not getting appropriate attention, and one of the reasons for that could be that it is a new repo and people are no longer subscribed to new repos automatically. So I'm going to ping some people in order to make sure that everyone is aware of this.

Please keep in mind that if you don't react with :eyes: your name won't be added to the list of reviewers, and in future you won't be notified about soon-to-be-merged proposals and your confirmation won't be required (which is totally fine if you're not interested in this, but if you're actively working on something related to the perl 6 project it's better to be on the list to make sure that upcoming changes will not disturb your efforts in unwanted ways).

@Benabik @FCO @FROGGS @Infinoid @MasterDuke17 @Mouq @TimToady @Util @arnsholt @bdw @bingos @clarkema @coke @colomon @cxreg @diakopter @gerdr @hankache @hoelzro @ingydotnet @japhb @jdhore @jnthn @kjs @kyleha @labster @lizmat @masak @mathw @mberends @moritz @niner @nunorc @particle @patrickas @peschwa @pmichaud @pmurias @retupmoca @samcv @sergot @sorear @stmuk @szabgab @tadzik @tbrowder @tene @usev6 @zhuomingliang

**Comment by AlexDaniel** [comment]
@ugexe I have thought about the terminology, but I don't think that “subject-matter expert” or “domain expert” describes people on these lists accurately. Of course, to be on that list, you most likely need to be an SME, but in my opinion that's not enough. There could be many rakudo experts, but in my view only one should be providing the direction where things should be going. Of course, they are free to delegate PRs, and they can also add more people if that is wanted. But just knowing a lot about a subject shouldn't be a reason to be added to the list. Does that make sense?

That being said, I'd love to use some existing term for that.

**Comment by AlexDaniel** [comment]
@zoffixznet, by any chance, can you take a look at it? What do you think, does it solve any problems? Or, more importantly, what does it *not* solve that needs to be solved?

**Comment by AlexDaniel** [comment]
Current plan: look at how things go with https://github.com/perl6/problem-solving/issues/3, make adjustments to the doc if needed and move to the review step on this PR asap.

**Comment by AlexDaniel** [comment]
OK, there are already [some issues](https://github.com/perl6/problem-solving/issues) and it looks like we'll get even more once we merge this. Let's get this thing going then.

**Comment by AlexDaniel** [comment]
About 7 days left to leave your reviews. Without any opposition it will go as is :)
@JJ @masak @timo @jnthn @ugexe @MasterDuke17 @samcv

File: README.md [hunk 2]
```diff
@@ -1 +1,167 @@
...
+
+If there are common principles that are used for a particular label,
+assignees are welcome to compose a document in order to avoid
+repeating themselves.
+
+Note that all people (even non-devs) who have enough experience in
+affected areas are welcome to provide feedback. That being said, the
+assignee has the final word on decisions in the issue/PR, so they are
+free to engage in discussions as they see fit. In rare cases they can
+moderate/lock the discussion if so is unavoidable.
```
Comment (jnthn) [inline README.md]: if *doing* so is unavoidable

**Comment by AlexDaniel** [comment]
It has been 15 days now, therefore merging it now. Thanks!

**Pull request merged**